lib/archethic_web/explorer/live/dashboard_live.ex

```diff
@@ -4,147 +4,166 @@ defmodule ArchethicWeb.Explorer.DashboardLive do
-          |> Enum.into(%{})
-      end
-
-    new_socket =
-      socket
-      |> update(:stats, &Map.put(&1, public_key, new_data))
-
-    {:noreply, new_socket}
+  def stats_to_nodes(stats) do
+    stats
+    |> Enum.reduce(MapSet.new(), fn {{node, _datetime}, _durations}, acc ->
+      MapSet.put(acc, node)
+    end)
+    |> MapSet.to_list()
   end
 
-  def render(assigns) do
-    View.render(DashboardView, "dashboard.html", assigns)
+  defp fetch_stats() do
+    # DashboardMetricsAggregator returns an _UNORDERED_ map
+    # we convert it to a ORDERED list of pairs
+    DashboardMetricsAggregator.get_all()
+    |> Enum.map(fn {{node_public_key, datetime}, duration_by_address} ->
+      {{Base.encode16(node_public_key), datetime}, duration_by_address}
+    end)
+    |> Enum.sort_by(fn {{_, datetime}, _} -> datetime end, {:asc, DateTime})
   end
 
-  defp aggregate(
-         acc,
-         {"archethic_mining_full_transaction_validation_duration", %{count: count, sum: sum}}
-       ) do
-    {acc_count, acc_sum} =
-      get_in(acc, [Access.key(:cumul, %{}), :transaction_validation]) || {0, 0}
-
-    new_count = count - acc_count
-    new_tps = new_count / 5
-
-    new_duration =
-      if new_count > 0 do
-        (sum - acc_sum) / new_count
-      else
-        0
-      end
-
-    acc
-    |> Map.put(:nb_transactions, new_count)
-    |> Map.put(:tps, Float.round(new_tps, 2))
-    |> Map.put(:validation_duration, new_duration)
+  defp network_transactions_count(stats) do
+    stats
+    |> Enum.reduce(%{}, fn {{_node, datetime}, duration_by_address}, acc ->
+      addresses_set =
+        duration_by_address
+        |> Enum.map(&elem(&1, 0))
+        |> MapSet.new()
+
+      Map.update(
+        acc,
+        DateTime.to_unix(datetime),
+        addresses_set,
+        &MapSet.union(&1, addresses_set)
+      )
+    end)
+    |> Enum.map(fn {timestamp, addresses_set} ->
+      {timestamp, MapSet.size(addresses_set)}
+    end)
+    |> Enum.into(%{})
   end
 
-  defp aggregate(acc, _), do: acc
+  defp network_transactions_avg_duration(stats) do
+    stats
+    |> Enum.reduce(%{}, fn {{_node, datetime}, duration_by_address}, acc ->
+      durations =
+        duration_by_address
+        |> Enum.map(&elem(&1, 1))
+
+      cur_count = length(durations)
+      cur_sum = Enum.sum(durations)
+
+      Map.update(
+        acc,
+        DateTime.to_unix(datetime),
+        %{count: cur_count, sum: cur_sum},
+        fn %{count: count, sum: sum} ->
+          %{count: count + cur_count, sum: sum + cur_sum}
+        end
+      )
+    end)
+    |> Enum.map(fn
+      {timestamp, %{count: 0, sum: 0}} ->
+        {timestamp, 0}
 
-  defp cumul(
-         acc,
-         {"archethic_mining_full_transaction_validation_duration", %{count: count, sum: sum}}
-       ) do
-    put_in(acc, [Access.key(:cumul, %{}), :transaction_validation], {count, sum})
+      {timestamp, %{count: count, sum: sum}} ->
+        {timestamp, div(sum, count) / 1_000_000}
```

Comment:
Why do you divide by 1_000_000, this result in 0.00035 ms on the graph which is invalid